apps/expo-go/ios/Exponent/Kernel/DevSupport/EXPackagerLogHelper.swift

```diff
@@ -0,0 +1,145 @@
+// Copyright 2015-present 650 Industries. All rights reserved.
+
+import Foundation
+import React
+
+@objc enum EXLogLevel: Int {
+  case info
+  case warning
+  case error
+}
+
+@objc class EXPackagerLogHelper: NSObject, SRWebSocketDelegate {
+  private var connected: Bool = false
+  private var socket: SRWebSocket?
+  private var bundleURL: URL
+  private var pendingMessage: String?
+  private var logLevel: EXLogLevel
+  private var onComplete: (() -> Void)?
+
+  @objc init(bundleURL: URL, level: EXLogLevel) {
+    self.bundleURL = bundleURL
+    self.logLevel = level
+  }
+
+  deinit {
+    if let socket = socket, connected {
```

Comment:
should we also cleanup socket even in disconnected state? given this is in the deinit state

Reply:
Something like this?
```suggestion
    socket?.close()
    self.socket = nil
    connected = false
```